### ISSUE TITLE
Plugin fails in 3.3.0-Beta

### ISSUE THREAD

**Issue by BerndJM** [markwal/Cura-OctoPrintUpload#6]
Hi Mark,
looks like they changed again something in the upcoming 3.3.0 version :-(((
Hope this are the important lines from the logfile:
```
2018-04-09 16:57:45,524 - ERROR - [MainThread] UM.Logger.logException [81]: Exception: Unable to write to file overhangtest3d: 'BuildPlateModel' object has no attribute 'activeBuildPlate'
2018-04-09 16:57:45,527 - ERROR - [MainThread] UM.Logger.logException [85]: Traceback (most recent call last):
2018-04-09 16:57:45,529 - ERROR - [MainThread] UM.Logger.logException [85]:   File "/build/3.3/build/inst/lib/python3.5/site-packages/UM/Qt/Bindings/OutputDeviceManagerProxy.py", line 122, in _writeToDevice
2018-04-09 16:57:45,531 - ERROR - [MainThread] UM.Logger.logException [85]:   File "/home/bernd/.local/share/cura/3.3/plugins/Cura-OctoPrintUpload/OctoPrintOutputDevice.py", line 72, in requestWrite
2018-04-09 16:57:45,535 - ERROR - [MainThread] UM.Logger.logException [85]:     active_build_plate_id = Application.getInstance().getBuildPlateModel().activeBuildPlate
2018-04-09 16:57:45,537 - ERROR - [MainThread] UM.Logger.logException [85]: AttributeError: 'BuildPlateModel' object has no attribute 'activeBuildPlate'
```
If I can provide further information: let me know.

**Comment by probonopd** [comment]
Possibly I am running into the same issue in https://github.com/probonopd/WirelessPrinting/issues/30.

**Comment by markwal** [comment]
Yes, they've changed the interface to get the gcode_list again.  If I fix it now, the plugin will be broken for people still on 3.2.

**Comment by BerndJM** [comment]
No idea how much work this will be, but what's about a seperate branch for 3.3 ? So the current "master" stays unchanged.
I wish I could help you with this, but my Python programming skills  are nearly "not existing" :-(

**Comment by markwal** [comment]
It's a good idea.  A branch for betas.

**Comment by BerndJM** [comment]
Meanwhile the 3.3 is stable, no longer Beta. 
So no more surprises are to expect ('til the 3.4) ;-))

**Comment by markwal** [comment]
Ah good.  I should patch this then.

**Comment by BerndJM** [comment]
Had some spare time this weekend and poked a bit in the code, think I found what has changed.
In OctoPrintOutputDevice.py change line 72 from
`active_build_plate_id = Application.getInstance().getBuildPlateModel().activeBuildPlate`
to
`active_build_plate_id = Application.getInstance().getMultiBuildPlateModel().activeBuildPlate`

No idea if that is all or there are side-effects, but for me it works.

**Comment by BerndJM** [comment]
Can confirm that it works (with this little fix) also with the 3.4Beta (and so hopefully with the 3.4 Final too).
Made a pull request for it.

**Comment by markwal** [comment]
master branch compatibility is now with Ultimaker Cura v3.6.  Before or after that and it won't load.
